### ISSUE TITLE
SHARE_BOX_FEATURE_EXTRACTOR  does not work

### ISSUE THREAD

**Issue by qianyizhang** [facebookresearch/maskrcnn-benchmark#761]
It seems SHARE_BOX_FEATURE_EXTRACTOR does not work at all since box_head predictor and mask_head predictor expects different shape tensor.

More specifically, box_feature_extractor ('FPN2MLPFeatureExtractor' and 'FPNXconv1fcFeatureExtractor') uses view to flatten its spatial resolution and adds some fc afterwards, results in tensor shape of NC.
While 'MaskRCNNFPNFeatureExtractor' uses a series of convs which results in tensor shape of NCHW

Furthermore, lets assume we make a common shared feature_extractor and moves the flattening/pooling + fc to box_head's predictor. I think mask_head should then set its own extractor to None and reusing the feature from box_head completely, instead of 
https://github.com/facebookresearch/maskrcnn-benchmark/blob/1127bdd368613f320f7b113320e62994c0baa216/maskrcnn_benchmark/modeling/roi_heads/mask_head/mask_head.py#L66

Another question in regard the implementation:
https://github.com/facebookresearch/maskrcnn-benchmark/blob/1127bdd368613f320f7b113320e62994c0baa216/maskrcnn_benchmark/modeling/roi_heads/box_head/box_head.py#L56
https://github.com/facebookresearch/maskrcnn-benchmark/blob/1127bdd368613f320f7b113320e62994c0baa216/maskrcnn_benchmark/modeling/roi_heads/box_head/loss.py#L132-L133
Is there any reason to wrap and immediately unwrap the tensor into list? Or it's simply obsolete formatting

**Comment by fmassa** [comment]
`SHARE_BOX_FEATURE_EXTRACTOR` only works without FPN (so the C4 models), and is present only to support C4 models, which use the same head.

The wrapping and unwrapping of `class_logits` and `box_regression`, as you point out, is a historical artifact that hasn't been cleaned up before the release.
